R/args.R

```diff
@@ -147,15 +149,38 @@ SampleArgs <- R6::R6Class(
       self$thin <- thin
       self$max_depth <- max_depth
       self$metric <- metric
-      # self$metric_file <- character()
+      self$inv_metric <- inv_metric
+      if(!is.null(inv_metric)) {
+        if(!is.null(metric_file)) {
+          stop("Only one of inv_metric and metric_file can be non-NULL")
+        }
+
+        # wrap inv_metric in list if not in one
+        if(!is.list(inv_metric)) {
+          inv_metric = list(inv_metric)
+        }
+
+        # write all inv_metrics to disk
+        inv_metric_paths <- c()
+        for(metric_i in 1:length(inv_metric)) {
+          inv_metric_path <-
+            file.path(
+              cmdstan_tempdir(),
+              paste0("inv_metric-", metric_i, ".json")
+            )
```

Comment:
I just fixed a bug (that was my fault) on master in https://github.com/stan-dev/cmdstanr/commit/c2badbe8d18ddf3278dd0ac20b956e3b699b3c88 that's related to this. We need to call `tempfile` in situations like this to get some randomness in the name to avoid overwriting in the case that someone uses the same model object with different (inv_)metrics to create separate fit objects. The corresponding fix here would be: 

```r
inv_metric_path <-
  tempfile(
   pattern = paste0("inv_metric-", metric_i, "-"),
   tmpdir = cmdstan_tempdir(),
   fileext = ".json"
  )
```